### PR TITLE
Update missing docs on `activation_dropout` and fix DropOut docs for SEW-D

### LINKED ISSUE #25854: `hidden_dropout` is not used in SEW-D
### System Info

NA

### Who can help?

@sanchit-gandhi

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Doing a quick Ctrl+F in https://github.com/huggingface/transformers/blob/main/src/transformers/models/sew_d/modeling_sew_d.py reveals that `config.hidden_dropout` is never used. Instead, `config.activation_dropout` is used in places where `config.hidden_dropout` is used in Wav2Vec2. Furthermore, `config.activation_dropout` is not even documented (https://github.com/huggingface/transformers/blob/main/src/transformers/models/sew_d/configuration_sew_d.py, Wav2Vec2 config also does not document `config.activation_dropout`).

To be more specific, for Wav2Vec2, `config.activation_dropout` is only used in the middle of the MLP block

https://github.com/huggingface/transformers/blob/07998ef39926b76d3f6667025535d0859eed61c3/src/transformers/models/wav2vec2/modeling_wav2vec2.py#L655

`config.hidden_dropout` is used at the end of MLP block, end of attention block, after adding positional embeddings

https://github.com/huggingface/transformers/blob/07998ef39926b76d3f6667025535d0859eed61c3/src/transformers/models/wav2vec2/modeling_wav2vec2.py#L664
https://github.com/huggingface/transformers/blob/07998ef39926b76d3f6667025535d0859eed61c3/src/transformers/models/wav2vec2/modeling_wav2vec2.py#L685
https://github.com/huggingface/transformers/blob/07998ef39926b76d3f6667025535d0859eed61c3/src/transformers/models/wav2vec2/modeling_wav2vec2.py#L761

Meanwhile, for SEW-D, there is no dropout in the middle of MLP block, and `config.activation_dropout` is used for end of MLP block, end of attention block, and positional embeddings in DeBERTa attention

https://github.com/huggingface/transformers/blob/07998ef39926b76d3f6667025535d0859eed61c3/src/transformers/models/sew_d/modeling_sew_d.py#L676
https://github.com/huggingface/transformers/blob/07998ef39926b76d3f6667025535d0859eed61c3/src/transformers/models/sew_d/modeling_sew_d.py#L725
https://github.com/huggingface/transformers/blob/07998ef39926b76d3f6667025535d0859eed61c3/src/transformers/models/sew_d/modeling_sew_d.py#L966

I guess the reason for this is due to translating from the original implementation, where DeBERTa's `hidden_dropout_prob` is fixed to `activation_dropout`

https://github.com/asappresearch/sew/blob/5c66f391bf137b12d4860c4cde24da114950b0fe/sew_asapp/models/squeeze_wav2vec2_deberta.py#L98

Nevertheless, this creates discrepancy between similar models i.e. `hidden_dropout`/`attention_dropout` means different things for Wav2Vec2 and SEW-D (and documentation on the configuration is not accurate). More importantly, when using SEW-D, if we increase `hidden_dropout`, nothing changes. Just by reading the documentation alone, which does not mention `activation_dropout`, there is no way to increase dropout for SEW-D layers.

### Expected behavior

I propose replacing `config.activation_dropout` with `config.hidden_dropout` in the SEW-D code. We would need to update various SEW-D configs also, since from what I see, the configs use `activation_dropout=0` and `hidden_dropout=0.1`.

### PR DESCRIPTION
**Pull request by gau-nernst**
# What does this PR do?

Fixes #25854

Add doc for `activation_dropout` for various audio models. Let me know if I miss out any.

For SEW-D, document the behavior that `hidden_dropout` is not used, while `activation_dropout` acts more like `hidden_dropout` in other models.

On a side note, it will be good if there is a test to catch undocumented config attributes in the future.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sanchit-gandhi

### CODE REVIEW

File: src/transformers/models/sew_d/configuration_sew_d.py [hunk 1]
```diff
@@ -72,6 +72,8 @@ class SEWDConfig(PretrainedConfig):
             The non-linear activation function (function or string) in the encoder and pooler. If string, `"gelu"`,
             `"relu"`, `"selu"`, `"gelu_python"` and `"gelu_new"` are supported.
         hidden_dropout (`float`, *optional*, defaults to 0.1):
+            Not used.
```
Comment (stevhliu) [inline src/transformers/models/sew_d/configuration_sew_d.py:75]: If `hidden_dropout` is not used, maybe we can just remove it? cc @amyeroberts

**Review by stevhliu — APPROVED** [review]
Good catch, thanks for adding these!

File: src/transformers/models/sew_d/configuration_sew_d.py (hunk 1 above)
Comment (amyeroberts) [reply src/transformers/models/sew_d/configuration_sew_d.py:75]: We'd need to do a deprecation cycle in case any users use it in their own code. So something along the lines of: 


```py
class SEWDConfig(PretrainedConfig):
    """
    ...
    hidden_dropout (`float`, *optional*, defaults to 0.1):
        Deprecated. Not used by the model and will be removed in a future version. 
    ...
    """
    def __init__(...):
        ...
        self._hidden_dropout = hidden_dropout

    @property 
    def hidden_dropout(self): 
        logger.warning_once("hidden_dropout is not used by the model and will be removed as config attribute in v4.35")
        return self._hidden_dropout

    def to_dict(self):
        """
        Serializes this instance to a Python dictionary.
        """
        output = super().to_dict()
        output["hidden_dropout"] = output.pop("_hidden_dropout")
        return output    
```

Comment (gau-nernst) [reply src/transformers/models/sew_d/configuration_sew_d.py]: @amyeroberts I added as you suggested.

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing these!

File: src/transformers/models/sew_d/configuration_sew_d.py [hunk 2]
```diff
@@ -280,3 +282,16 @@ def __init__(
     @property
     def inputs_to_logits_ratio(self):
         return functools.reduce(operator.mul, self.conv_stride, 1)
+
+    @property
+    def hidden_dropout(self):
```
Comment (sanchit-gandhi) [inline src/transformers/models/sew_d/configuration_sew_d.py:287]: Nice!

**Review by sanchit-gandhi — COMMENTED** [review]
Thanks for the clean PR @gau-nernst!
